visualize-admin/visualization-tool
# refactor: Move dataSet from configurator state to chart state

This PR provides the technical foundation for the dashboards to allow charts based on different datasets within the same configurator state.

Essentially, it moves the `dataSet` property from the configurator state down to the charts themselves.

### How to test
1. Go to the deployment preview.
2. Add the following object to the local storage (use `vizualize-configurator-state:EpXqRKzyJxIy` key):
```
{"version":"3.0.0","state":"CONFIGURING_CHART","dataSource":{"type":"sparql","url":"https://lindas.admin.ch/query"},"meta":{"title":{"de":"","fr":"","it":"","en":""},"description":{"de":"","fr":"","it":"","en":""}},"chartConfigs":[{"key":"tPfhn1Lsv4u1","version":"2.3.0","meta":{"title":{"en":"","de":"","fr":"","it":""},"description":{"en":"","de":"","fr":"","it":""}},"dataSet":"https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/9","chartType":"column","filters":{"https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/Kanton":{"type":"single","value":"https://ld.admin.ch/canton/1"}},"interactiveFiltersConfig":{"legend":{"active":false,"componentIri":""},"timeRange":{"active":false,"componentIri":"https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/Jahr","presets":{"type":"range","from":"","to":""}},"dataFilters":{"active":false,"componentIris":[]},"calculation":{"active":false,"type":"identity"}},"fields":{"x":{"componentIri":"https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/Jahr","sorting":{"sortingType":"byAuto","sortingOrder":"asc"}},"y":{"componentIri":"https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/AnzahlAnlagen"}}},{"key":"tWem-qDvBFyn","version":"2.3.0","meta":{"title":{"en":"","de":"","fr":"","it":""},"description":{"en":"","de":"","fr":"","it":""}},"dataSet":"https://environment.ld.admin.ch/foen/ubd0104/6","chartType":"line","filters":{"https://environment.ld.admin.ch/foen/ubd0104/parametertype":{"type":"single","value":"E.coli"},"https://environment.ld.admin.ch/foen/ubd0104/station":{"type":"single","value":"https://environment.ld.admin.ch/foen/ubd0104/Station/CH10001"}},"interactiveFiltersConfig":{"legend":{"active":false,"componentIri":""},"timeRange":{"active":false,"componentIri":"https://environment.ld.admin.ch/foen/ubd0104/dateofprobing","presets":{"type":"range","from":"","to":""}},"dataFilters":{"active":false,"componentIris":[]},"calculation":{"active":false,"type":"identity"}},"fields":{"x":{"componentIri":"https://environment.ld.admin.ch/foen/ubd0104/dateofprobing"},"y":{"componentIri":"https://environment.ld.admin.ch/foen/ubd0104/value"}}}],"activeChartKey":"tPfhn1Lsv4u1"}
```
3. Go to `/en/create/EpXqRKzyJxIy?dataSource=Prod`.
4. See that the first chart is based on Photovoltaikanlagen dataset, while the second one is based on Bathing water quality dataset.
